Feature/subscriptions (#875)
mercadopago/devsite-docs: guides/online-payments/subscriptions/integration.en.md

`auto_recurring.frequency_type` (required) | Indicates the type of frequency. It can be by month (months) or day (days). <br><br> Along with the frequency, they define the installment cycle that a subscription will have. <br><br>For example, if every fifteen days you need to generate an instalment to be charged it would look like this: `auto_recurring.frequency`: 15 y `auto_recurring.frequency_type`: days |
`auto_recurring.transaction_amount` | If we indicate the amount, it is fixed. If this field is blank, it is understood as a variable amount. A maximum of two decimals separated by a period is allowed. |
`auto_recurring.currency_id` (required) | Identifies the currency that corresponds to the country. |
`auto_recurring.repetitions` | Indicates if the subscription will have a limit. If not specified, there is no limit. This limit is related to `auto_recurring.frequency` y `auto_recurring.frequency_type`. |
`auto_recurring.free_trial.frequency` | Defines an initial test period and delays the first collection. Indicates the length of time that the service will not be charged, for based on the type of frequency. It must be consistent with `auto_recurring.frequency`. |

> `auto_recurring.free_trial.frequency_type` | Indicates the number of installments that will not be charged for the service. It must be consistent with `auto_recurring.frequency_type` |

Reviewer: ```suggestion
`auto_recurring.free_trial.frequency_type` | Indicates the number of installments that will not be charged for the service. It must be consistent with `auto_recurring.frequency_type`. |
```

Attribute |	Definition
--- | ---
`reason` | It is the description that the subscriber will see when making the subscription and the detail that will be seen in the card statement. |
`status` | Subscription status. It can be `pending` or `authorized`. |
`auto_recurring.frequency` | Indicates the length of time or cycle based on the type of frequency. |
`auto_recurring.frequency_type` | Indicates the type of frequency. It can be by month (months) or day (days). Along with the frequency, they define the installment cycle that a subscription will have.<br><br> For example, if every fifteen days you need to generate an instalment to be collected it would look as follows: `auto_recurring.frequency`: 15 y  `auto_recurring.frequency_type`: days |
`auto_recurring.transaction_amount` | Amount applicable to the subscription. |
`auto_recurring.currency_id` | Identifies the currency that corresponds to the country. |
`auto_recurring.end_date` | Indicates if the subscription will have a limit. If not specified, there is no limit. |
`auto_recurring.free_trial.frequency` | Indicates the length of time that the service will not be charged. It must be consistent with `auto_recurring.frequency`. |

> `auto_recurring.free_trial.frequency_type` | Indicates the amount of installments that will not be charged for the service. It must be consistent with `auto_recurring.frequency_type` |

Reviewer: ```suggestion
`auto_recurring.free_trial.frequency_type` | Indicates the amount of installments that will not be charged for the service. It must be consistent with `auto_recurring.frequency_type`. |
```

---
indexable: false
---

# Integrate subscriptions  

## Integration types

There are two ways to integrate subscriptions: 

* __With an associated plan__: Use this form when you need to use the same subscription on different occasions and want to organize them into identifiable groups. For example, for a monthly and annual subscription to a gym.  

* __Without an associated plan__: Use this form when you know that different subscriptions will not share any features because they will be very specific or specialized for each payer. For example, for a single month subscription with a special discount.

> NOTE
> 
> Key concepts
> 

> > Do you have questions about what a plan or other concept is? Keep the <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/guides/online-payments/subscriptions/introduction" target="_blank">key concepts</a> handy to review them when needed.

Reviewer 2: ```suggestion
> Do you have questions about what a plan or other concept is? Keep the <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/guides/online-payments/subscriptions/introduction" target="_blank">key concepts</a> handy to review them when needed.
```

If you want to use a subscription with an associated plan, you must first create a plan.  

### Create plan

When generating the plan you will get the `preapproval_plan_id` that you will use to make the subscription. 

To create the plan, make the following call to our API with the data you need:

[[[
```curl curl --location --request POST 'https://api.mercadopago.com/preapproval_plan' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer ENV_ACCESS_TOKEN' \
--data-raw '{
	"back_url":"https://www.mercadopago[FAKER][URL][DOMAIN]",
	"reason":"Plan Pase Mensual Gold",
	"auto_recurring":{
		"frequency":"1",
		"frequency_type":"months",
        "transaction_amount":1100,
		"currency_id":"[FAKER][CURRENCY][ACRONYM]",
		"repetitions":12,
		"free_trial":{
			"frequency_type":"months",
			"frequency":"1"
		}
	}
}'
```
]]]

#### Attributes

Attribute |	Definition
--- | ---
`reason` (required) | It is the description that the subscriber will see when making the subscription and the detail that will be seen in the card statement. |
`auto_recurring.frequency` (required) | Indicates the length of time  or cycle based on the type of frequency. |

> `auto_recurring.frequency_type` (required) | Indicates the type of frequency. It can be by month (months) or day (days).

Reviewer: ```suggestion
`auto_recurring.frequency_type` (required) | Indicates the type of frequency. It can be by month (months) or day (days). <br><br> Along with the frequency, they define the installment cycle that a subscription will have. <br><br>For example, if every fifteen days you need to generate an instalment to be charged it would look like this: `auto_recurring.frequency`: 15 y `auto_recurring.frequency_type`: days |
```

> Along with the frequency, they define the installment cycle that a subscription will have. <br><br>For example, if every fifteen days you need to generate an instalment to be charged it would look like this: `auto_recurring.frequency`: 15 y `auto_recurring.frequency_type`: days |

Reviewer: ```suggestion
```

        "frequency_type": "months",
        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "repetitions": 12,
        "free_trial": {
            "frequency": 1,
            "frequency_type": "months"
        }
    }
}

```
Done! We can now create the subscription and associate it to your plan.

### Create subscription

Once you have generated your plan and obtained your `preapproval_plan_id`, create the subscription of the payer by API as follows: 

[[[
```curl curl --location --request POST 'https://api.mercadopago.com/preapproval \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer ENV_ACCESS_TOKEN' \
--data-raw '{
   "preapproval_plan_id":"2c938084726e18d60172720000000000",
   "card_token_id":"9fca87c7338585abd000111222333444",
   "payer_email":"test_user_XXXX@testuser.com"
   
}'
```
]]]

#### Attributes

Attribute |	Definition
--- | ---
`preapproval_plan_id` (required) | Refers to the previously generated plan.|
`card_token_id` (required) | The information on the card will be converted into a token to send the data securely. |
`payer_email` (required) | Payer's email address. |

> WARNING
> 
> Important
> 

> > ¿Do you have questions about how to create the payment token? Find all the information in the section of __<a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/guides/sdks/official/js/" target="_blank">Capture data from the card</a>__.

Reviewer: ```suggestion
> ¿Do you have questions about how to create the payment token? Find all the information in the section of __<a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/guides/online-payments/checkout-api/receiving-payment-by-card#bookmark_card_data_capture" target="_blank">Capture data from the card</a>__.
```

> > ¿Do you have questions about how to create the payment token? Find all the information in the section of __<a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/guides/sdks/official/js/" target="_blank">Capture data from the card.</a>__.

Reviewer 2: ```suggestion
> ¿Do you have questions about how to create the payment token? Find all the information in the section of __<a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/guides/sdks/official/js/" target="_blank">Capture data from the card.</a>__.
```

#### Response 
`HTTP Status 200 OK`
```json
{
    "id": "2c938084726e18d60172750000000000",
    "preapproval_plan_id": "2c938084726e18d60170001112223334",
    "payer_id": 100200300,
    "payer_email": "test_user_XXXX@testuser.com",
    "back_url": "http://www.mercadopago[FAKER][URL][DOMAIN]/",
    "collector_id": 10101,
    "application_id": 1234567812345678,
    "status": "authorized",
    "reason": "Plan Pase Mensual Gold",
    "external_reference": "125124513",
    "date_created": "2020-06-02T08:37:42.734-04:00",
    "last_modified": "2020-06-02T08:37:42.735-04:00",
    "init_point":  "https://www.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726e18d60172750000000000",
    "sandbox_init_point": "https://sandbox.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726e18d60172750000000000",
    "auto_recurring": {
        "frequency": 1,
        "frequency_type": "months",
        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "start_date": "2020-07-02T08:37:42.734-04:00",
        "end_date": "2021-07-02T11:59:52.581-04:00"
}
```

> >You can get more information about the fields in the <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/reference/" target="_blank">API Reference</a>.

Reviewer 2: ```suggestion
>You can get more information about the fields in the <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/reference/" target="_blank">API Reference</a>.
```

  "reason": "Suscripción Pase Mensual Gold - Particular",
  "status": "pending"
}'
```
]]]

#### Response
`HTTP Status 200 OK`
```json

{
    "id": "2c938084726fca480172750000000000",
    "payer_id": 400500600,
    "payer_email": "test_user_XXXX@testuser.com",
    "back_url": "http://www.mercadopago[FAKER][URL][DOMAIN]/",
    "collector_id": 100200300,
    "application_id": 1234567812345678,
    "status": "pending",
    "reason": "Suscripción Pase Mensual Gold - Particular",
    "external_reference": "23546246234",
    "date_created": "2020-06-02T09:07:14.260-04:00",
    "last_modified": "2020-06-02T09:07:14.263-04:00",
    "init_point": "https://www.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726fca480172750000000000",
    "sandbox_init_point": "https://sandbox.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726fca480172750000000000",
    "auto_recurring": {
        "frequency": 1,
        "frequency_type": "months",
        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "start_date": "2020-06-02T09:07:14.260-04:00",
        "end_date": "2022-07-20T11:59:52.581-04:00"
    },
    "version": 0
}
```

> > You can get more information about the fields in the <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/reference/" target="_blank">API reference</a>.

Reviewer 2: ```suggestion
> You can get more information about the fields in the <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/reference/" target="_blank">API reference</a>.
```

Attributes

Attribute |	Definition
--- | ---
`reason` | It is the description that the subscriber will see when making the subscription and the detail that will be seen in the card statement. |
`status` | Subscription status. It can be `pending` or `authorized`. |
`auto_recurring.frequency` | Indicates the length of time or cycle based on the type of frequency. |

> `auto_recurring.frequency_type` | Indicates the type of frequency. It can be by month (months) or day (days).

Reviewer: ```suggestion
`auto_recurring.frequency_type` | Indicates the type of frequency. It can be by month (months) or day (days). Along with the frequency, they define the installment cycle that a subscription will have.<br><br> For example, if every fifteen days you need to generate an instalment to be collected it would look as follows: `auto_recurring.frequency`: 15 y  `auto_recurring.frequency_type`: days |
```

> Along with the frequency, they define the installment cycle that a subscription will have.<br><br> For example, if every fifteen days you need to generate an instalment to be collected it would look as follows: `auto_recurring.frequency`: 15 y  `auto_recurring.frequency_type`: days |

Reviewer: ```suggestion
```

`auto_recurring.transaction_amount (required)` | Amount applicable to the subscription. |
`auto_recurring.currency_id (required)` | Identifies the currency that corresponds to the country. |
`auto_recurring.end_date` | Indicates if the subscription will have a limit. If not specified, there is no limit. |
`auto_recurring.free_trial.frequency` | Indicates the length of time that the service will not be charged. It must be consistent with `auto_recurring.frequency` |
`auto_recurring.free_trial.frequency_type` | Indicates the amount of installments that will not be charged for the service. It must be consistent with `auto_recurring.frequency_type` |
`collector_id (required)` | Seller's identifier. |
`payer_email (required)` | Payer's email address. |
`card_token_id (required)` | If the subscription was already authorized, the information on the card will be converted into a token to send the data securely. |

------------
### Next steps

> LEFT_BUTTON_RECOMMENDED_EN
>
> Tests
>
> Check that your subscriptions are properly configured with the test users. 
>

> > [Pruebas](http://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/guides/online-payments/subscriptions/testing/)

Reviewer 2: ```suggestion
> [Pruebas](http://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/guides/online-payments/subscriptions/testing/)
```

> RIGHT_BUTTON_RECOMMENDED_EN
>
> Advanced integration
>
> Update, edit or cancel your subscriptions. 
>

> > [Pruebas](http://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/guides/online-payments/subscriptions/advanced-integration/)

Reviewer 2: ```suggestion
> [Pruebas](http://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/guides/online-payments/subscriptions/advanced-integration/)
```
